src/cascadia/TerminalApp/AppActionHandlers.cpp

```diff
@@ -1264,6 +1264,54 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    void TerminalPage::_HandleSaveTask(const IInspectable& /*sender*/,
+                                       const ActionEventArgs& args)
+    {
+        if (Feature_SaveTask::IsEnabled())
+        {
+            if (args)
+            {
+                if (const auto& realArgs = args.ActionArgs().try_as<SaveTaskArgs>())
+                {
+                    if (realArgs.Commandline().empty())
+                    {
+                        if (const auto termControl{ _GetActiveControl() })
+                        {
+                            if (termControl.HasSelection())
+                            {
+                                const auto selections{ termControl.SelectedText(true) };
+                                const auto selection = std::accumulate(selections.begin(), selections.end(), std::wstring());
+                                realArgs.Commandline(selection);
+                            }
+                        }
+                    }
+
+                    try
+                    {
+                        winrt::Microsoft::Terminal::Control::KeyChord keyChord = nullptr;
+                        if (!realArgs.KeyChord().empty())
+                        {
+                            keyChord = KeyChordSerialization::FromString(winrt::to_hstring(realArgs.KeyChord()));
+                        }
+                        _settings.GlobalSettings().ActionMap().AddSendInputAction(realArgs.Name(), realArgs.Commandline(), keyChord);
+                        _settings.WriteSettingsToDisk();

```

Comment:
❌ `WriteSettingsToDisk` writes the file on _this_ thread. We definitely can't be doing this on the UI thread. 

You might be able to just do a 

```c++
auto saveSettings = [settings=_settings]() -> winrt::fire_and_forget {
  co_await winrt::resume_background();
  settings.WriteSettingsToDisk();
};
saveSettings();
```

to basically make a lambda that does the fire_and_forget thing, but then also just call it immediately. I haven't tested it, so I'm sure there's _spooky coroutine reasons_ that's a bad idea, but 🤷

Reply:
Also now that I'm looking at it, the Settings UI itself saves the settings on the UI thread! Egads! I don't think it's worth blocking you over this if we can't even do it right

Reply:
I gave this a try just so I could get a better understanding of how the fire_and_forget stuff works.

It looked like it worked,  the settings were saved to the file, and I was able to see it switch from the window thread to the background thread in the debugger when it did the write to disk.

In the debugger I started getting a AV.  Adding a strong ref to settings seemed to prevent that.
```c++
auto saveSettings = [settings = _settings]() -> winrt::fire_and_forget {
    const auto strongSettings = settings;
    co_await winrt::resume_background();
    strongSettings.WriteSettingsToDisk();
};
saveSettings();
```